### ISSUE TITLE
Gurumin: A Monstrous Adventure (322290) - often crashes

### ISSUE THREAD

**Issue by unit73e** [ValveSoftware/Proton#2177]
# Compatibility Report
- Name of the game with compatibility issues: Gurumin: A Monstrous Adventure
- Steam AppID of the game: 322290

## System Information
- GPU: R9 380
- Driver/LLVM version: Mesa 18.3.1
- Kernel version: 4.20
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/unit73e/3ce983e30e0120a5f15e7a2db270283b
- Proton version: 3.16-6 Beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

This is similar to the following issue: https://github.com/ValveSoftware/Proton/issues/2171

However this issue didn't occur anymore and I'm not sure why. Apparently I incorrectly assumed it was because of issues on my vulkan installation but the game does not use vulkan. I'm not sure if this is because the game is a Direct3D 9 game and Vulkan is runs on Direct3D 10/11 games. If possible I'd like to know if this is the case so that I can input better issue reports.

[steam-322290.log](https://github.com/ValveSoftware/Proton/files/2719479/steam-322290.log)

## Symptoms
Crashes in certain parts of the game.

## Reproduction
Go through the door in the Potato Ruins level.
![20190101025230_1](https://user-images.githubusercontent.com/11358308/50574799-9b8fa680-0de7-11e9-9efd-13ae2b0d95ce.jpg)

There are probably other parts of the game that crash but this one is fairly consistent.

**Comment by unit73e** [comment]
I ran the game side by side and have pin pointed the problem to these lines:

```
2844.913:0008:0031:fixme:dsound:IDirectSoundBufferImpl_Restore (0x81a8c50):stub
2845.042:0008:0031:fixme:dsound:IDirectSoundBufferImpl_Restore (0x81a8ae0):stub
2845.313:0008:0031:fixme:dsound:IDirectSoundBufferImpl_Restore (0x81a8ae0):stub
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2845.411:0008:002c:err:d3d:wined3d_debug_callback 0x2f8cc38: "GL_INVALID_OPERATION in glVertexAttribPointer(non-VBO array)".
2847.988:0026:0027:trace:module:LdrShutdownProcess ()
```

Unfortunately I'm not experienced with OpenGL so I have no clue what that means or if there's a way to workaround the issue. Either way it seems to me like either the game is poorly programmed or the driver is being picky, but I could be totally wrong.

**Comment by unit73e** [comment]
Adding the `LIBGL_ALWAYS_SOFTWARE=1` will still show the errors but at least the game does not crash. Note that since I'm running Mesa I can use these variables: https://www.mesa3d.org/envvars.html

**Comment by unit73e** [comment]
This issue seems to be fixed with version Proton 4.2

EDIT: or maybe amdgpu latest drivers? Either it didn't crash in an area that was always crashing.
